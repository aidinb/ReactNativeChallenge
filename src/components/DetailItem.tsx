import React from 'react';
import { View, Text, StyleSheet, Dimensions } from 'react-native';
import {COLORS} from '../styles';

const { width } = Dimensions.get('window');

interface DetailItemProps {
  title?: string;
  value?: string;
}

const DetailItem: React.FC<DetailItemProps> = ({ title, value }) => {
  if (!title || !value) {
    return null;
  }

  return (
      <View style={styles.detailItem}>
        <Text style={styles.title}>{title}</Text>
        <Text style={styles.value}>{value}</Text>
      </View>
  );
};

const styles = StyleSheet.create({
  detailItem: {
    width,
    paddingVertical: 15,
    borderBottomWidth: 1,
    borderBottomColor: COLORS.gray,
    paddingHorizontal: 10,
  },
  title: {
    fontSize: 14,
    color: COLORS.darkGray,
    marginBottom: 5,
  },
  value: {
    fontSize: 16,
    color: COLORS.black,
    textAlign: 'center',
  },
});

export default DetailItem;
