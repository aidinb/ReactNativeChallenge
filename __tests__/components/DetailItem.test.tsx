import React from 'react';
import { render } from '@testing-library/react-native';
import DetailItem from '../../src/components/DetailItem';

describe('DetailItem Component', () => {
    it('renders correctly with title and value', () => {
        const { toJSON, getByText } = render(
            <DetailItem title="Test Title" value="Test Value" />
        );

        // Check if the component renders correctly
        expect(toJSON()).toMatchSnapshot();

        // Check if title and value are displayed
        expect(getByText('Test Title')).toBeTruthy();
        expect(getByText('Test Value')).toBeTruthy();
    });

    it('does not render anything if title is missing', () => {
        const { toJSON } = render(<DetailItem title="" value="Test Value" />);

        // Ensure component returns null
        expect(toJSON()).toBeNull();
    });

    it('does not render anything if value is missing', () => {
        const { toJSON } = render(<DetailItem title="Test Title" value="" />);

        // Ensure component returns null
        expect(toJSON()).toBeNull();
    });

    it('does not render anything if both title and value are missing', () => {
        const { toJSON } = render(<DetailItem title="" value="" />);

        // Ensure component returns null
        expect(toJSON()).toBeNull();
    });
});
