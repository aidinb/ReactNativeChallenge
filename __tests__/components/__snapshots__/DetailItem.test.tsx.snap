// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailItem Component renders correctly with title and value 1`] = `
<View
  style={
    {
      "borderBottomColor": "gray",
      "borderBottomWidth": 1,
      "paddingHorizontal": 10,
      "paddingVertical": 15,
      "width": 750,
    }
  }
>
  <Text
    style={
      {
        "color": "#888",
        "fontSize": 14,
        "marginBottom": 5,
      }
    }
  >
    Test Title
  </Text>
  <Text
    style={
      {
        "color": "black",
        "fontSize": 16,
        "textAlign": "center",
      }
    }
  >
    Test Value
  </Text>
</View>
`;
